lib/test.rb

```diff
@@ -420,38 +420,69 @@ def setup_formulae_deps_instances(formula, formula_name)
         return
       end
 
-      build_dependents_from_source = %w[
+      build_dependents_from_source_whitelist = %w[
         cabal-install
         docbook-xsl
         ghc
         go
         ocaml
+        ocaml-findlib
+        ocaml-num
         openjdk
         rust
-      ].include?(formula_name)
+      ]
 
       uses_args = []
       uses_args << "--recursive" unless Homebrew.args.skip_recursive_dependents?
-      # whitelist specific formula where we want to test build dependants.
-      uses_args << "--include-build" if build_dependents_from_source
       dependents =
-        Utils.popen_read("brew", "uses", "--include-test", *uses_args, formula_name)
+        Utils.popen_read("brew", "uses", "--include-build", "--include-test", *uses_args, formula_name)
```

Comment:
We cannot take this simply from a top level view. We need to consider it from a dependent level.

Why? Consider this example:

A is on the whitelist and builds dependent B
C is not on the whitelist but also wants to test dependent B.

B should be deferred (skipped) from the A build, but when C builds dependents it should still build B from source - something that wouldn't be done if we only consider it on the top level of A & C.